addons/website/models/ir_http.py

```diff
@@ -87,17 +88,21 @@ def _auth_method_public(cls):
             super(Http, cls)._auth_method_public()
 
     @classmethod
-    def _extract_website_page(cls, response):
-        if getattr(response, 'status_code', 0) != 200:
+    def _register_website_track(cls, response):
+        if getattr(response, 'status_code', 0) != 200 or not hasattr(response, 'qcontext'):
             return False
-
-        main_object = getattr(response, 'qcontext', {}).get('main_object')
-        return main_object if getattr(main_object, '_name', False) == 'website.page' else False
+        main_object = response.qcontext.get('main_object')
+        website_page = main_object if getattr(main_object, '_name', False) == 'website.page' else False
+        template = response.qcontext.get('response_template')
+        view = request.env['website'].get_template(template) if template else False
+        url = request.httprequest.url
```

Comment:
move into the if 2 line later

Reply:
do we really need it, can't the _handle_webpage_dispatch got it from request directly ?